File: src/main/java/com/github/guikeller/jettyrunner/util/IconUtil.java
package com.github.guikeller.jettyrunner.util;

import javax.swing.*;
import java.net.URL;

/**
 * IconUtil - Loads the jetty icon
 * @author Gui Keller
 */
public class IconUtil {

    private static final IconUtil INSTANCE = new IconUtil();
    private static final Icon ICON = loadIcon();

    private IconUtil(){
        super();
    }

    public static IconUtil getInstance(){
        return INSTANCE;
    }

    private static Icon loadIcon() {
        URL resource = INSTANCE.getClass().getResource("/jetty-icon.png");
        return new ImageIcon(resource);
    }

    public Icon getIcon(){
        return ICON;
    }

}

File: src/main/kotlin/com/github/guikeller/jettyrunner/util/IconUtil.kt
package com.github.guikeller.jettyrunner.util

import javax.swing.Icon
import javax.swing.ImageIcon

/**
 * IconUtil - Loads the jetty icon
 * @author Guy Keller
 */
class IconUtil private constructor() {

    companion object {
        private var instance: IconUtil? = null
        fun getInstance(): IconUtil {
            if (instance == null) {
                instance = IconUtil()
            }
            return instance as IconUtil
        }
    }

    private var icon: Icon? = null

    private fun loadIcon(): Icon {
        if (icon == null) {
            val resource = javaClass.getResource("/META-INF/jetty-icon.png")
            icon = ImageIcon(resource)
        }
        return icon as Icon
    }

    fun getIcon() = icon ?: loadIcon()

}
